[docs] Write a full release post for v0.31.0 (#999)
cozystack/cozystack: docs/changelogs/v0.31.0.md

Cozystack v0.31.0 is a significant release that brings new features, key fixes, and updates to underlying components.
This version enhances GPU support, expands automation (like autoscaling and CI/CD), and introduces a more robust release process for greater stability .
Below we outline the highlights in each area for current users and developers.

## Major Features and Improvements

### GPU support for tenant Kubernetes clusters

Cozystack now integrates NVIDIA GPU Operator support for tenant Kubernetes clusters.

> This allows platform users to run GPU-enabled applications such as AI/ML in their own clusters.

Reviewer: ```suggestion
This allows platform users to run GPU-powered applications for AI/ML in their own clusters.
```

Author: AI/ML are not the only applications using GPU, I guess.

This paves the road to running Cozystack on ARM64 servers.
Changes include Makefile improvements (https://github.com/cozystack/cozystack/pull/907)
and multi-arch Docker image builds (https://github.com/cozystack/cozystack/pull/932 and https://github.com/cozystack/cozystack/pull/970).

We thank Nikita Bykov @nbykov0 for his ongoing work on ARM support!

<!--
* Introduce support for cross-architecture builds and Cozystack on ARM:
    * [build] Refactor Makefiles introducing build variables. (@nbykov0 in https://github.com/cozystack/cozystack/pull/907)
    * [build] Add support for multi-architecture and cross-platform image builds. (@nbykov0 in https://github.com/cozystack/cozystack/pull/932 and https://github.com/cozystack/cozystack/pull/970)
-->

### VerticalPodAutoscaler (VPA) Expansion

The VerticalPodAutoscaler is now enabled for more Cozystack components to automate resource tuning.
Specifically, VPA was added for tenant Kubernetes control planes (@klinch0 in https://github.com/cozystack/cozystack/pull/806),
the Cozystack Dashboard (https://github.com/cozystack/cozystack/pull/828),
and the Cozystack etcd-operator (https://github.com/cozystack/cozystack/pull/850).

> All Cozystack components which have VPA enabled can automatically adjust their CPU/memory requests based on usage, improving resource efficiency.

Reviewer: ```suggestion
All Cozystack components which have VPA enabled can automatically adjust their CPU/memory requests based on usage, improving platform stability.
```
or
```suggestion
All Cozystack components which have VPA enabled can automatically adjust their CPU/memory requests based on usage, improving application stability.
```

Author: So it's more about not dying to resource misconfiguration, than to saving pennies?

* [postgres] Remove duplicated `template` entry from backup manifest. (@etoshutka in https://github.com/cozystack/cozystack/pull/872)
* [kube-ovn] Fix versions mapping in Makefile. (@kvaps in https://github.com/cozystack/cozystack/pull/883)
* [dx] Automatically detect version for migrations in the installer.sh. (@kvaps in https://github.com/cozystack/cozystack/pull/837)
* [dx] remove version_map and building for library charts. (@kvaps in https://github.com/cozystack/cozystack/pull/998)
* [docs] Review the tenant Kubernetes cluster docs. (@NickVolynkin in https://github.com/cozystack/cozystack/pull/969)
* [docs] Explain that tenants cannot have dashes in the names. (@NickVolynkin in https://github.com/cozystack/cozystack/pull/980)

## Dependencies

> * MetalLB is now included directly as a patched image based on version 0.14.9. (@lllamnyp in https://github.com/cozystack/cozystack/pull/945)

Reviewer: ```suggestion
* MetalLB images are now built in-tree based on version 0.14.9 with additional critical patches. (@lllamnyp in https://github.com/cozystack/cozystack/pull/945)
```

Author: fixed

* Update KamajiControlPlane to edge-25.4.1. (@kvaps in https://github.com/cozystack/cozystack/pull/953, fixed by @nbykov0 in https://github.com/cozystack/cozystack/pull/983)
* Update cert-manager to v1.17.2. (@kvaps in https://github.com/cozystack/cozystack/pull/975)

## Development, Testing, and CI/CD

### Testing

Improvements:

> * Introduce `cozytest` - a new testing framework by @kvaps in https://github.com/cozystack/cozystack/pull/982

Author: didn't know about the implementation details, thank you

Fixes:

> * Increase timeout durations for `capi` and `keycloak` to improve reliability during environment setup. (@kvaps in https://github.com/cozystack/cozystack/pull/858)

Author: fixed
